### PR TITLE
DOC-1583 - RS: Create certificate updates (self-signed & TLS guidelines)

### PR DESCRIPTION
**Pull request by lanceleonard**
- Initial draft

### CODE REVIEW

**Comment by lanceleonard** [comment]
A staged preview [is available](https://docs.redis.com/staging/jira-doc-1583/rs/security/certificates/create-certificates/).

File: content/rs/security/certificates/create-certificates.md [hunk 1]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
 aliases: 
 ---
 
-Follow these instructions to create your own certificates to install on your Redis Enterprise cluster. Note that you can install a separate certificate per cluster component.
+When you first install Redis Enterprise Software, self-signed certificates are created to encrypt internal traffic.  These certificates expire after a year (365 days) and must be renewed.
 
-1. Create a private key:
+You can renew these certificates by replacing them with new self-signed certificates or by replacing them with certificates signed by a certificate authority (CA).
```
Comment (rrelledge) [inline content/rs/security/certificates/create-certificates.md:13]:
```suggestion
You can renew these certificates by replacing them with new self-signed certificates or by replacing them with certificates signed by a [certificate authority (CA)](https://en.wikipedia.org/wiki/Certificate_authority).
```

File: content/rs/security/certificates/create-certificates.md [hunk 2]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+| `cm` | The admin console |
+| `metrics_exporter` | The metrics exporter |
+| `proxy` | The database endpoint |
+| `syncer` | The synchronization process |
+
+You can also use the REST API.  To learn more, see [Update certificates]({{<relref "/rs/security/certificates/updating-certificates#how-to-update-certificates">}}).
+
+## Create CA-signed certificates
+
+You can use certificates signed by a certificate authority (CA).
```
Comment (rrelledge) [inline content/rs/security/certificates/create-certificates.md:121]:
```suggestion
You can use certificates signed by a [certificate authority (CA)](https://en.wikipedia.org/wiki/Certificate_authority).  
```

**Review by rrelledge — APPROVED** [review]
Looks good to me. I left a couple of suggestions to consider.

File: content/rs/security/certificates/create-certificates.md (hunk 2 above)
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed.

File: content/rs/security/certificates/create-certificates.md [hunk 3]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+    ```
+    _Important:&nbsp;_ The .CNF file is a configuration file.  Check with your security team or certificate authority for help creating a valid configuration file for your environment.
+
+3.  Sign the private key using your certificate authority.
+
+    The signing process varies for each organization and CA vendor.  Consult your security team and certificate authority for specific instructions describing how to sign a certificate.
+
+4.  Upload the certificate to your cluster.
+
+    You can use `rladmin` to replace the existing certificates with new certificates:
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed.

File: content/rs/security/certificates/create-certificates.md [hunk 4]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+    ``` bash
+    $ openssl genrsa -out <key-file-name>.pem 2048
     ```
 
-    Replace the following variables with your own values:
-    
-    - `<cert-name>` - The name of the certificate to update. See the [certificates table]({{<relref "/rs/security/certificates">}}) for the list of valid certificate names.
-    - `<cert-file-name>` - The certificate filename 
-    - `<key-file-name>` - The key filename
+1.  Create certificate signing request.
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed.

File: content/rs/security/certificates/create-certificates.md [hunk 5]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
 
-    To upload the new certificate and replace the current certificate with the [`rladmin`]({{<relref "/rs/references/cli-utilities/rladmin">}}) command-line utility,
-    run the [`cluster certificate set`]({{<relref "/rs/references/cli-utilities/rladmin/cluster/certificate">}}) command:
+    Individual operating systems might limit access to specific algorithms.  (For example, Ubuntu 20.04 [limits  access](https://manpages.ubuntu.com/manpages/focal/man7/crypto-policies.7.html) to <nobr>SHA-1</nobr>.)  In such cases, Redis Enterprise Software is limited to the features supported by the underlying operating system.
 
-    ```sh
-    rladmin cluster certificate set <cert-name> \
-        certificate_file <cert-file-name>.pem \
-        key_file <key-file-name>.pem
+### Creating certificates
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed

File: content/rs/security/certificates/create-certificates.md [hunk 6]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+    dns=*.internal.<cluster-fqdn>
     ```
 
-    {{<note>}}
-You will be prompted for a Country Name, State or Province Name, Locality Name, Organization Name, Organizational Unit, and Common Name.
+1.  The Extended Key Usage attribute should be set to `TLS Web Client Authentication` and `TLS Web Server Authentication`.
+
+1.  We strongly recommend using a strong hash algorithm, such as <nobr>SHA-256</nobr> or <nobr>SHA-512</nobr>.
+
+    Individual operating systems might limit access to specific algorithms.  (For example, Ubuntu 20.04 [limits  access](https://manpages.ubuntu.com/manpages/focal/man7/crypto-policies.7.html) to <nobr>SHA-1</nobr>.)  In such cases, Redis Enterprise Software is limited to the features supported by the underlying operating system.
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed

File: content/rs/security/certificates/create-certificates.md [hunk 7]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+
+1.  Set the following values according to the values specified by your security team or certificate authority:
+
+    - Country Name
+    - State or Province Name
+    - Locality Name
+    - Organization Name
+    - Organization Unit 
+
+1.  The Subject Alternate name (SAN) should include the following values based on the FQDN:
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed.

File: content/rs/security/certificates/create-certificates.md [hunk 8]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+
+1.  Use the cluster's fully qualified domain name (FQDN) as the certificate Common Name (CN).
+
+1.  Set the following values according to the values specified by your security team or certificate authority:
+
+    - Country Name
+    - State or Province Name
+    - Locality Name
+    - Organization Name
+    - Organization Unit
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed.

File: content/rs/security/certificates/create-certificates.md [hunk 9]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+
+### TLS certificate guidelines
+
+When you create certificates signed by a certificate authority, you need to create server certificates and client certificates.  The follow sections describe guidelines that apply to both certificates and provide guidance for each certificate type. 
+
+#### Guidelines for server and client certificates
+
+1.  Include the full [certificate chain](https://en.wikipedia.org/wiki/X.509#Certificate_chains_and_cross-certification) when creating certificate .PEM files for either server or client certificates.   
+
+1.  Certificates in the .PEM file should be chained in the following order:
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Reworked to remove passive voice.

File: content/rs/security/certificates/create-certificates.md [hunk 10]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+
+## Create CA-signed certificates
+
+You can use certificates signed by a certificate authority (CA).  
+
+For best results, use the following guidelines to create the certificates.
+
+### TLS certificate guidelines
+
+When you create certificates signed by a certificate authority, you need to create server certificates and client certificates.  The follow sections describe guidelines that apply to both certificates and provide guidance for each certificate type.
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed

File: content/rs/security/certificates/create-certificates.md [hunk 11]
```diff
@@ -8,45 +8,222 @@ categories: ["RS"]
...
+``` text
+mycluster.example.com
+*.mycluster.example.com
+anothercluster.example.com 
+*.anothercluster.example.com
+```
+
+### Step 2: Locate the new certificate files
+
+When successful, the script generates two .PEM files for each generated certificate: A certificate file and certificate key, each named after the type of certificate generated (see earlier table for individual certificate names.)
```
[the remark this answers is not in the document]
Comment (lanceleonard) [reply content/rs/security/certificates/create-certificates.md]: Fixed

**Pull request merged**